coral/src/app/features/approvals/schemas/components/schema-request-table.tsx

```diff
@@ -0,0 +1,101 @@
+import { Box, DataTableColumn, GhostButton, Icon } from "@aivenio/aquarium";
```

Comment:
Nitpick: I'm  not super convinced by stashing this type and the table columns separately in this file. The name of the file is also a little confusing to me. The snake case looks like it is be like the `zod` schemas for forms, but it it not "pure" config, as columns can (and do) also return JSX elements. Also, we define the `rows` inline in the `SchemaApprovalsTable`, so we have the columns and rows in two different spots 🤔 

Maybe this could be moved to `SchemaApprovalTable`?

Reply:
🙈  Yes, I see. 

I mainly moved it outside since that enables me (better) to use `columns` in the tests. And with "enables me better" I mean "not exporting something out of the `SchemaApprovalTable` component just for the sake of testing it -> which I'm still doing, only not as visible 🙈 🙈 🙈 🙈 

I can move it inside `SchemaApprovalTable` again 😬 I still would export it from there for testing purpose - I usually don't like doing that as it's often a bad pattern, in this case I feel it improves testability (or rather: the quality of tests) a lot, which I would vote higher then encapsulation.

WDYT?

Reply:
>by checking that the correct things are rendered according to the columns 🤔
True, it does cover a lot of internals implicitly, but at the same time it covers a lot of the things we need to know (are all columns there, right headers, right links etc.) in an easy way and without having additional sets of data (like header names, correct link names, correct formatted dates). 

While we don't need to test that a function passed to formatter does actually format stuff (that's tested in DS), we need to make sure that we get the _right_ formatting, because that's the functionality we're passing. Same for buttons etc. 

I'm a bit torn about his - having additional sets of data means that for example changing a header name in the component will make the test fail -> that can feel annoying, since it's not functionality that is broken, but a mismatch in text. I'm personally ok'ish with that, since I see it more as a assertion that all texts are correct & it also makes the test itself more readable. But also more work to write and maintain.

Using the columns to test makes the test faster to write, but if we're doing something wrong in components (which we can, e.g. pass nonsense formatting), our tests would happily run green and we would not be any wiser :D

Reply:
> While we don't need to test that a function passed to formatter does actually format stuff (that's tested in DS), we need to make sure that we get the right formatting, because that's the functionality we're passing. Same for buttons etc.

Absolutely, but this can be done on individual cells rendered, right? 

> I'm a bit torn about his - having additional sets of data means that for example changing a header name in the component will make the test fail -> that can feel annoying, since it's not functionality that is broken, but a mismatch in text. I'm personally ok'ish with that, since I see it more as a assertion that all texts are correct & it also makes the test itself more readable. But also more work to write and maintain.

yeah, this is for sure very annoying. But then again, I'm not sure how it's possible without risking what you describe next:

> Using the columns to test makes the test faster to write, but if we're doing something wrong in components (which we can, e.g. pass nonsense formatting), our tests would happily run green and we would not be any wiser :D

I think the "grabage in garbage out" problem can also be avoided with doing assessment on single cells, with a hardcoded value of what we expect in the test 🤔 

Also, the code in here actually requirews the columns to be declared inline in the body of the component that renders them, so we can't even export them 😬  https://github.com/aiven/klaw/pull/611

Reply:
To conclude: I'll update the tests and add `columns` to the component!